lib/std/zig/system/x86.zig

```diff
@@ -0,0 +1,664 @@
+const std = @import("std");
+const Target = std.Target;
+const CrossTarget = std.zig.CrossTarget;
+
+fn setFeature(cpu: *Target.Cpu, feature: Target.x86.Feature, enabled: bool) void {
+
+    const idx = @as(Target.Cpu.Feature.Set.Index, @enumToInt(feature));
+
+    if(enabled) cpu.features.addFeature(idx)
+    else cpu.features.removeFeature(idx);
+}
+
+fn hasFeature(cpu: *Target.Cpu, feature: Target.x86.Feature) bool {
+    const idx = @as(Target.Cpu.Feature.Set.Index, @enumToInt(feature));
+    return cpu.features.isEnabled(idx);
+}
+
+inline fn bit(input: u32, offset: u5) bool {
+    return (input >> offset) & 1 != 0;
+}
+
+pub fn detectNativeCpuAndFeatures(cpu: *Target.Cpu) void {
+
+    defer {
+        // Whenever we find a model, add that model's featureset.
+        cpu.features.addFeatureSet(cpu.model.features);
+    }
+
+    // When we can't identify a specific model,
+    // we guess based on processor features.
+    // This seems to be the accepted standard.
```

Comment:
Generally, if a CPU model cannot be determined accurately, then I would prefer it to choose a generic CPU model. There's a [generic one for i386](https://github.com/ziglang/zig/blob/1cf3209cb8c03227987b6f4fe43838bf224c4f26/lib/std/target/x86.zig#L1748) and a [generic one for x86_64](https://github.com/ziglang/zig/blob/1cf3209cb8c03227987b6f4fe43838bf224c4f26/lib/std/target/x86.zig#L2654). This is modeled fine since the "features" field is the true source of information of what features are available, rather than the CPU model's features.